handler.js

```diff
@@ -0,0 +1,125 @@
+    const file_name_no_ext = path.basename(input_file_name, input_file_ext);
+    return file_name_no_ext + configData['output-extension'];
+};
+
+const getConfigData = () => {
+    const config = {
+        "input-bucket": process.env.INPUT_BUCKET,
+        "input-prefix": process.env.INPUT_PREFIX || "",
+        "output-bucket": process.env.OUTPUT_BUCKET || process.env.INPUT_BUCKET,
+        "output-prefix": process.env.OUTPUT_PREFIX || "results/",
+        "output-extension": process.env.OUTPUT_EXTENSION || "_results.json",
+        "output-enabled": process.env.OUTPUT_ENABLED || true,
+    }
+
+    return config;
+};
+
+async function runSaf(command_string) {
+    if (!command_string) {
+        throw new Error("SAF CLI Command String argument is required.");
+    }
+    
+    const saf_command = command_string.split(' ');
+    
+    const allowable_topics = ['convert', 'generate', 'harden', 'scan', 'validate', 'view'];
+    const topic = saf_command[0].split(':')[0];
+    
+    if (!allowable_topics.includes(topic)) {
+        throw new Error("The command string did not include one of the allowable topics: " + allowable_topics.join(', ') + ". Please reference the documentation for more details.");
+    }
+    
+    const command = saf_command[0].split(':')[1];
+    
+    if (topic == "view" & command == "heimdall") {
+        throw new Error("The SAF Action does not support the 'view heimdall' command. Please reference the documentation for other uses.");
+    }
+    
+    await saf.run(saf_command);
+}
+
+module.exports.saf = async (event, context, callback) => {
+    const configData = getConfigData();
+    const logger = createWinstonLogger(context.awsRequestId, process.env.LOG_LEVEL || 'debug');
+    logger.debug("Called SAF lambda function.");
+
+    const bucket = event.Records[0].s3.bucket.name;
+    const key = decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, " "));
+    
+    logger.info("Getting object with bucket: " + bucket + " and key: " + key);
+    const s3BucketObjectContents = await getObject(bucket, key);
+    
+    // TODO: Explore the saf update to use stdin instead of a file for the contents
+    const input_file_name = getInputFileName(key);
+    let INPUT_FILE = path.resolve('/tmp/', input_file_name);
+    logger.info("Input file: " + INPUT_FILE);
+    await fs.writeFileSync(INPUT_FILE, s3BucketObjectContents);
+
+    const command_string_input = process.env.COMMAND_STRING;
+    let command_string = `${command_string_input} -i ${INPUT_FILE}`;
+
+    const output_file_name = getOutputFileName(input_file_name, configData);
+    let OUTPUT_FILE = path.resolve('/tmp/', output_file_name);
+    if(configData['output-enabled']) {
+        command_string = `${command_string_input} -i ${INPUT_FILE} -o ${OUTPUT_FILE}`;
+    }
+    
+    logger.info("Calling SAF CLI with the command: " + command_string);
+    await runSaf(command_string)
+        .then(() => {
```

Comment:
there's no need to combine the async/await syntax with the promise syntax like this.  just 'await' the runsaf command, and then run whatever you need to afterwards

Reply:
Removed